pkg/build/api/types.go

```diff
@@ -188,6 +188,11 @@ const (
 	// StatusReasonExceededRetryTimeout is an error condition when the build has
 	// not completed and retrying the build times out.
 	StatusReasonExceededRetryTimeout = "ExceededRetryTimeout"
+
+	// StatusReasonMissingSecret indicates that the build is missing required
+	// secret for pushing the output image. The build will stay in the pending
+	// state until the secret is not created or the build timeout.
+	StatusReasonMissingSecret = "MissingSecret"
```

Comment:
@bparees it would be nice to re-use it, do you want to create special reason for each secret or have one generic?